operator_ui/src/components/Bridges/Form.tsx

```diff
@@ -1,42 +1,63 @@
-import React from 'react'
-import PropTypes from 'prop-types'
-import { Prompt } from 'react-router-dom'
-import * as formik from 'formik'
-import { withStyles } from '@material-ui/core/styles'
-import { TextField, Grid } from '@material-ui/core'
+import { Grid, TextField } from '@material-ui/core'
+import {
+  createStyles,
+  Theme,
+  withStyles,
+  WithStyles
+} from '@material-ui/core/styles'
 import Button from 'components/Button'
-import { set, get } from 'utils/storage'
+import formik, { FormikProps } from 'formik'
 import normalizeUrl from 'normalize-url'
+import React from 'react'
+import { Prompt } from 'react-router-dom'
+import { get, set } from 'utils/storage'
 
-const styles = theme => ({
-  textfield: {
-    paddingTop: theme.spacing.unit * 1.25
-  },
-  card: {
-    paddingBottom: theme.spacing.unit * 2
-  },
-  button: {
-    marginTop: theme.spacing.unit * 3
-  },
-  flash: {
-    textAlign: 'center',
-    paddingTop: theme.spacing.unit,
-    paddingBottom: theme.spacing.unit
-  }
-})
+const styles = (theme: Theme) =>
+  createStyles({
+    textfield: {
+      paddingTop: theme.spacing.unit * 1.25
+    },
+    card: {
+      paddingBottom: theme.spacing.unit * 2
+    },
+    button: {
+      marginTop: theme.spacing.unit * 3
+    },
+    flash: {
+      textAlign: 'center',
+      paddingTop: theme.spacing.unit,
+      paddingBottom: theme.spacing.unit
+    }
+  })
 
-const isDirty = ({ values, submitCount }) => {
+const isDirty = ({ values, submitCount }: Props) => {
   return (
     (values.name !== '' ||
       values.url !== '' ||
-      (values.minimumContractPayment.toString() !== '0' &&
-        values.confirmations !== '0')) &&
+      (values.minimumContractPayment !== '0' && values.confirmations !== 0)) &&
     submitCount === 0
   )
 }
 
-const Form = props => (
-  <React.Fragment>
+interface OwnProps extends Partial<FormValues>, WithStyles<typeof styles> {
+  actionText: string
+  onSubmit: any
```

Comment:
Type this as a Function or `() => void`?

Reply:
Oh good catch on these, I made them as `any` while refactoring and forgot to actual type them

Reply:
Just a heads up, decided I'll mark them as `CHECKME` and then type the event handlers after I type the functions that use `Form`